src/app/ui/shared/appoitment-form/appoitment-form.component.ts

```diff
@@ -0,0 +1,149 @@
+  });
+  medicServices: string[];
+  formTitle: string;
+  pacientName: string;
+  users: Observable<any>;
+  animals: AnimalUtilInfo[];
+  doctorAppointment: DoctorsAppointmentDTO;
+  appoinementFormPlaceHolder;
+  @ViewChild('pacientList') pacientList: any;
+  @ViewChild('inputPacientName') inputPacientName: any;
+
+  constructor(
+    private doctorService: DoctorService,
+    private servicesService: DoctorServicesService,
+    private activeModal: NgbActiveModal,
+    private userService: UserService,
+    private doctorAppointmentService: DoctorAppointmentsService,
+    private appointmentFormService: AppointmentFormService
+  ) {}
+
+  ngOnInit(): void {
+    this.appoinementFormPlaceHolder = appointmentFormData;
+    this.formTitle = appointmentFormData.title;
+    this.pacientName = '';
+
+    this.doctorService
+      .getDoctorById('o2Jt7YS9zCWvBfDWY08X')
+      .pipe(take(1))
+      .subscribe((medic: DoctorDTO) => {
+        this.sampleForm.patchValue({
+          medicName: medic.doctorName,
+          medLocation: medic.location,
+          medicId: 'o2Jt7YS9zCWvBfDWY08X',
+        });
+      });
+
+    this.servicesService
+      .getAllServices('o2Jt7YS9zCWvBfDWY08X')
+      .pipe(take(1))
+      .subscribe((docServices: DoctorServiceDTO) => {
+        this.medicServices = docServices[0].services;
+      });
+  }
+
+  onSubmit(): void {
+    const newAnimalInfo = new AnimalUtilInfo()
+      .setName(this.sampleForm.value.pacientAnimal.animalName)
+      .setUid(this.sampleForm.value.pacientAnimal.animalId);
+
+    const newDoctorAppointment = new DoctorsAppointmentDTO()
+      .setUserName(this.sampleForm.value.pacientName)
+      .setUserId(this.sampleForm.value.pacientId)
+      .setServices(this.sampleForm.value.medService)
+      .setDateTime(
+        this.formatDateAndTime(
+          this.sampleForm.value.startDate,
+          this.sampleForm.value.startTime
+        )
+      )
+      .setAnimal(newAnimalInfo)
+      .setLocation(this.sampleForm.value.medLocation);
+
+    this.doctorAppointmentService.createAppointment(
+      [newDoctorAppointment],
+      this.sampleForm.value.medicId
+    );
+    this.activeModal.close();
+  }
+
+  onCancelForm(): void {
+    this.activeModal.close();
+  }
+
+  filterPacients(searchText: string): void {
+    this.users = this.appointmentFormService.filterPacients(searchText, this.pacientName);
+  }
+
+  onSeclectPacient(selectedPacient: UserDto): void {
+    this.pacientList.nativeElement.classList.add('hidden');
+    this.sampleForm.patchValue({
+      pacientName: selectedPacient['name'],
+      pacientId: selectedPacient['id'],
+    });
+    this.animals = selectedPacient['animals'];
+  }
+
+  formatDateAndTime(date: any, time: any): string {
+    let hour = time.hour;
+    let month = date.month;
+    if (hour <= 9) {
+      hour = '0'.concat(hour);
+    }
+    if (month <= 9) {
+      month = '0'.concat(month);
+    }
+    return (
+      date.year +
```

Comment:
can put on the same row - its pretty hard to read it like this

Reply:
done